dashboard/app/models/code_review.rb

```diff
@@ -0,0 +1,71 @@
+# == Schema Information
+#
+# Table name: code_review_requests
+#
+#  id              :bigint           not null, primary key
+#  user_id         :integer          not null
+#  script_id       :integer          not null
+#  level_id        :integer          not null
+#  project_id      :integer          not null
+#  project_version :string(255)      not null
+#  closed_at       :datetime
+#  deleted_at      :datetime
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+# Indexes
+#
+#  index_code_review_requests_unique  (user_id,script_id,level_id,closed_at,deleted_at) UNIQUE
+class CodeReview < ApplicationRecord
+  # This model was renamed partway through the development process. This line
+  # will be removed when the table is renamed before this work is completed.
+  self.table_name = 'code_review_requests'
+
+  acts_as_paranoid
+
+  # TODO: Once all the renaming has settled, the following association should be:
+  # has_many :comments, class_name: 'CodeReviewComment', dependent:  :destroy
+  has_many :comments, class_name: 'CodeReviewNote', foreign_key: 'code_review_request_id', dependent:  :destroy
+
+  # Enforce that each student can only have one open code review per script and
+  # level. (This is also enforced at the database level with a unique index.)
+  validates_uniqueness_of :user_id, scope: [:script_id, :level_id],
+    conditions: -> {where(closed_at: nil)},
+    message: 'already has an open code review for this script and level'
+
+  # Returns an array of all code reviews (open and closed) that match the given
+  # identifying attributes. Returns an empty array if there are no matches.
+  def self.get_all(user_id:, script_id:, level_id:)
```

Comment:
Good eyes and good question!  Without changing the style of how we write controllers and models _too_ much (since we haven't formally adopted this direction), I wanted to start inching closer to the design described [here](https://kellysutton.com/2019/10/29/taming-large-rails-codebases-with-private-activerecord-models.html).  The TL;DR is that we should use as little of the ActiveRecord surface as possible outside of the model.  If this feels like it strays too far from our usual pattern, I can change it.